hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -674,14 +674,15 @@ private static HoodieLogBlock getBlock(HoodieWriteConfig writeConfig,
                                          List<HoodieRecord> records,
                                          boolean shouldWriteRecordPositions,
                                          Map<HeaderMetadataType, String> header,
-                                         String keyField) {
+                                         String keyField,
+                                         HoodieStorage storage) {
     switch (logDataBlockFormat) {
       case AVRO_DATA_BLOCK:
         return new HoodieAvroDataBlock(records, shouldWriteRecordPositions, header, keyField);
       case HFILE_DATA_BLOCK:
         return new HoodieHFileDataBlock(
             records, header, writeConfig.getHFileCompressionAlgorithm(), new StoragePath(writeConfig.getBasePath()),
-            writeConfig.getBooleanOrDefault(HoodieReaderConfig.USE_NATIVE_HFILE_READER));
+            writeConfig.getBooleanOrDefault(HoodieReaderConfig.USE_NATIVE_HFILE_READER), storage);
```

Comment:
Is passing the storage instance required here?  We should strictly do refactoring with changing as little argument as possible.

Reply:
I got rid of the change. Seems less efficient then just passing it though. We will see if it still passes ci.

Reply:
make inline stuff part of the storageconfig instead of storage

Reply:
We can change the logic in an independent PR.  It's usually good to avoid such argument passing changes if not necessary, so there's less risk of breaking the logic.